### DISCUSSION TITLE
Customizing main page

### DISCUSSION THREAD

**Discussion by iot-sle** [arendst/Tasmota#15183] [Q&A] [answered]
I use an ESP32 with many values recorded by sml sensor using 2 meters. So the original main page is not the best visualization, because the amount of values. A user web page with >w is not the best solution for me. So I use >W to change the main page.

Sml sensor allows to use * for hiding the recorded value from main page. Thats fine for me. Can I hide other values too?
By expample the DS1820 or the counters?

Another issue for me: I can't get the preceeding @ operator  for showing my web code on top of the main page working. I tried to use it at the beginning of every html line like @<style>  and also <@style>.  Any idea how to use it correctly?

```
>W
so(7)
<style>table {border-collapse:collapse;} table th,td {width:200px; align:right;} .linie {border-bottom:1px solid black;} </style>
<table>
<tr class="linie"></tr>
<tr><th>Gas       :</th><td>%0.2(Gas/100)% m³</td><td></td></tr>
<tr class="linie"><th>Energie Bezug EG:</th><td>%0.3(StromEG/1000)% kWh</td><td></td></tr>
<tr class="linie"><th> </th><td><b>ABB</td><td><b>ESY</td></tr>
<tr><th>ID:<td>%sml[11]%</td><td>%sml[1]%</td></tr> 
<tr><th>E Bezug :</th><td>%0.2sml[13]% kWh</td><td>%0.2sml[2]% kWh</td></tr>
<tr class="linie"><th>E Einsp.:</th><td>%0.2sml[14]% kWh</td><td>%0.2sml[3]% kWh</td></tr>
<tr class="linie"><th>E netto Gesamt:</th><td>%0.2sml[12]% kWh</td><td> </td></tr>

<tr><th>P     L1:</th><td>%0.2sml[16]% W</td><td>%0.2sml[5]% W</td></tr>
<tr><th>P     L2:</th><td>%0.2sml[17]% W</td><td>%0.2sml[6]% W</td></tr>
<tr class="linie"><th>P     L3:</th><td>%0.2sml[18]% W</td><td>%0.2sml[7]% W</td></tr>
<tr class="linie"><th>P Gesamt:</th><td>%0.2sml[15]% W</td><td>%0.2sml[4]%  W</td></tr>
<tr><th>U     L1:</th><td>%0.2sml[19]% V</td><td>%0.2sml[8]% V</td></tr>
<tr><th>U     L2:</th><td>%0.2sml[20]% V</td><td>%0.2sml[9]% V</td></tr>
<tr class="linie"><th>U     L3:</th><td>%0.2sml[21]% V</td><td>%0.2sml[10]% V</td></tr>
<tr class="linie"><th>Frequenz:</th><td>%0.2sml[22]% Hz</td>
</table>
```
Result:

![grafik](https://user-images.githubusercontent.com/74416643/159153866-67a1362d-2162-451b-b7b5-5b1bfdd2058e.png)

**Comment by sfromis** [comment]
The general concept for Tasmota is that the main page is not meant for "daily operation", but more as a place for admin/test, without trying to provide features for making web design. I can't answer about what you can do with scripting, but with ESP32, you also have the Berry language available, including features for web pages.

(You can get rid of the pretty useless "ESP32 Temperature" using `SetSensor127 0`)

**Reply by iot-sle** [reply]
Thanks for the tip about ESP temperature. I might try Berry some time later, when my projects are finished. I started sometime ago with rules and switched to scripting in between. At the moment I'm happy with that. >W section is fine for me, The only thing which I can't get work is the @operator.

**Comment by iot-sle** [comment, marked as answer]
I found the solution for hiding sensor values: WebSensor<x> 0 seems the right command.
With Websensor1 0; Websensor5; the counter and the ds1820 values are gone from the main page.
So the remaining question about how to use the @ operator is only for my interest. The screenshot is using text-align:right for td.

![grafik](https://user-images.githubusercontent.com/74416643/159160086-334b45c1-ac5f-4b95-88c3-f7c740adfa73.png)

The administrative button at the bottom of the page and the buttons for switches do now fit to the screen again, without giving up some values.

**Comment by gemu2015** [comment]
since you use so(7) you did set FORCE_PLAIN which disables the @ operator.
but since you use your own table structure the @ operator would not work anyhow.
because it simply puts the line into the predefined Tasmota table

**Comment by iot-sle** [comment]
Thank you for the answer.
